src/telemetry/metrics/serve.rs

```diff
@@ -180,3 +182,9 @@ impl Error for ServeError {
         }
     }
 }
+
+impl<F: FmtPrometheus> fmt::Display for PromWriter<F> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
This layer of indirection seemed unnecessary to me until I realized you couldn't `fmt::Display::fmt(metrics, &mut writer)` because `writer` is `io::Write` but not a `Formatter`. Maybe there ought to be a comment here somewhere?